core/src/main/java/com/es/core/dao/JdbcPhoneDao.java

```diff
@@ -146,17 +153,10 @@ private void add(final Phone phone) {
         setPhoneColors(phone);
     }
 
-    private String generateInConstraint(int n) {
-        StringBuilder result = new StringBuilder();
-        if (n > 0) {
-            result.append("where phones.id in (?");
-
-            for (int i = 0; i < n - 1; i++) {
-                result.append(",?");
-            }
-            result.append(")");
-        }
-        return new String(result);
+    private String generateInConstraint(List<Long> keys) {
```

Comment:
https://stackoverflow.com/questions/1327074/how-to-execute-in-sql-queries-with-springs-jdbctemplate-effectivly